server/controllers/user.js

```diff
@@ -0,0 +1,85 @@
+const {
+  findUser,
+  createUser,
+  comparePasswords,
+  createToken,
+  findAndDeleteUser,
+  findUserAndUpdate,
+} = require("../services/user");
+
+const handleError = (error) => {
+  let errors = {
+    email: error?.errors?.email?.message || "",
+    password: error?.errors?.password?.message || "",
+    firstname: error?.errors?.firstname?.message || "",
+    lastname: error?.errors?.lastname?.message || "",
+    dateOfBirth: error?.errors?.dateOfBirth?.message || "",
+  };
+  if (error.code === 11000) {
+    errors.email = "Email already exists";
```

Comment:
Why do you set text 'Email already exists' as email?

Reply:
what should it be?

Reply:
error is related to email field

Reply:
To be honest I dont like this. It's not validation issue. I'd like to throw exception with code and handle it in common exception handler

Reply:
can you show me an example for that?